lib/charliecloud.py

```diff
@@ -140,10 +140,10 @@ class Download_Mode(enum.Enum):
 ir_name: IR_PATH_COMPONENT
 ir_tag: ":" IR_TAG
 ir_digest: "@sha256:" HEX_STRING
-IR_HOST: /[A-Za-z0-9_.-]+/
-IR_PORT: /[0-9]+/
-IR_PATH_COMPONENT: /[a-z0-9_.-]+/
-IR_TAG: /[A-Za-z0-9_.-]+/
+IR_HOST: /[$A-Za-z0-9_.-]+/
+IR_PORT: /[$0-9]+/
+IR_PATH_COMPONENT: /[$a-z0-9_.-]+/
+IR_TAG: /[$A-Za-z0-9_.-]+/
```

Comment:
Suggest putting the `$` with the special characters at the end, so it doesn't look like a `$A` variable substitution. (Note the hyphen has to remain last so it's not misinterpreted as a range.)